pontoon/base/views.py

```diff
@@ -784,6 +785,53 @@ def translation_memory(request):
         return HttpResponse("no")
 
 
+def translation_memory_pg(request):
+    """
+    Suggests translations from internal translations.
+    """
+    try:
+        text = request.GET['text']
+        locale = request.GET['locale']
+        pk = request.GET['pk']
+    except MultiValueDictKeyError as e:
+        log.error(str(e))
+        return HttpResponse("error")
+
+    try:
+        locale = Locale.objects.get(code__iexact=locale)
+    except Locale.DoesNotExist as e:
+        log.error(e)
+        return HttpResponse("error")
+
+    min_quality = 0.7
+    max_results = 5
+    length = len(text)
+    min_dist = math.ceil(max(length * min_quality, 2))
+    max_dist = math.floor(min(length / min_quality, 1000))
+
+    # Only check entities with similar length
+    entries = TranslationMemoryEntry.objects.all().extra(
+        where=["(CHAR_LENGTH(source) BETWEEN %s AND %s)"\
+              "AND source %% %s"],
```

Comment:
@Osmose it's documented here: http://www.postgresql.org/docs/9.1/static/pgtrgm.html